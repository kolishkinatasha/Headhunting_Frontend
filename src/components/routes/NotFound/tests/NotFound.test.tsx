import { render, screen } from '@testing-library/react';
import React from 'react';
import renderer from 'react-test-renderer';

import NotFoundRoute from '../NotFound';

const renderComponent = () => render(<NotFoundRoute />);

describe('Компонент NotFound', () => {
  it('snapshot совпадает', () => {
    const tree = renderer.create(<NotFoundRoute />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('компонент отображается без ошибок', () => {
    expect(renderComponent).not.toThrow();
  });

  it('содержит нужный текст', () => {
    renderComponent();

    expect(screen.getByText(/Страница не найдена/i)).toBeInTheDocument();
  });
});
