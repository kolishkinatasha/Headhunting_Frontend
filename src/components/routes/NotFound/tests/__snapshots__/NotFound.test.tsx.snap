// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент NotFound snapshot совпадает 1`] = `
<h1>
  Страница не найдена.
</h1>
`;
